concrete/controllers/dialog/tree/node/file_folder/add.php

```diff
@@ -0,0 +1,104 @@
+<?php
+namespace Concrete\Controller\Dialog\Tree\Node\FileFolder;
+
+use Concrete\Controller\Dialog\Tree\Node;
+use Concrete\Core\Application\EditResponse;
+use Concrete\Core\Entity\File\StorageLocation\StorageLocation as StorageLocationEntity;
+use Concrete\Core\File\Filesystem;
+use Concrete\Core\File\StorageLocation\StorageLocationFactory;
+use Concrete\Core\Permission\Checker as Permissions;
+use Concrete\Core\Tree\Node\Type\FileFolder;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class Add extends Node
+{
+    protected $viewPath = '/dialogs/tree/node/file_folder/add';
+    protected $helpers = ['form', 'validation/token'];
+
+    protected function getNode()
+    {
+        if (!isset($this->node)) {
+            $filesystem = new Filesystem();
+            $treeNodeID = $this->app->make('helper/security')->sanitizeInt($this->request->query->get('treeNodeID'));
```

Comment:
@marvinde is there a reason why you use the `sanitizeInt` method, instead of `(int) $this->request->query->get('treeNodeID')`? (KISS)